nuxeo-services/nuxeo-platform-query-api/src/main/java/org/nuxeo/ecm/platform/query/nxql/NXQLQueryBuilder.java

```diff
@@ -421,16 +421,24 @@ protected static String atomicQueryElement(DocumentModel model, PredicateDefinit
                 // both min and max are not provided, ignore predicate
                 return "";
             }
-        } else if (operator.equals("IN")) {
+        } else if (operator.equals("IN") || operator.equals("NOT IN")) {
             List<String> options = getListValue(model, values[0]);
             if (options == null || options.isEmpty()) {
                 return "";
             } else if (options.size() == 1) {
-                return serializeUnary(parameter, "=", options.get(0));
+                if (operator.startsWith("NOT")) {
```

Comment:
Could you do the test using equality, to be faster? `if (operator.equals("IN")) {` (and inverse the two branches of the `if`)

Reply:
equals instead of startsWith : done
I don't inverse the two branches of the if to keep code homogeneous with the class